### PR TITLE
Fix various placement plan occupancy calculation bugs

### PR DESCRIPTION
**Pull request by Gekkio**

### CODE REVIEW

File: service/src/integrationTest/kotlin/fi/espoo/evaka/occupancy/OccupancyControllerIntegrationTest.kt [hunk 1]
```diff
@@ -231,14 +231,16 @@ class OccupancyControllerIntegrationTest : FullApplicationTest() {
...
+            preschoolDaycarePeriod = FiniteDateRange(LocalDate.of(2021, 6, 1), LocalDate.of(2021, 7, 31))
         )
 
         assertEquals(
             OccupancyResponseSpeculated(
                 max3Months = OccupancyValues(sum = 1.75, headcount = 1, caretakers = 1.0, percentage = 25.0),
                 max6Months = OccupancyValues(sum = 2.75, headcount = 2, caretakers = 1.0, percentage = 39.3),
-                max3MonthsSpeculated = OccupancyValues(sum = 2.75, headcount = 2, caretakers = 1.0, percentage = 39.3),
+                // preschool only
+                max3MonthsSpeculated = OccupancyValues(sum = 2.25, headcount = 2, caretakers = 1.0, percentage = 32.1),
```
Comment (jaahon) [inline service/src/integrationTest/kotlin/fi/espoo/evaka/occupancy/OccupancyControllerIntegrationTest.kt:242]: Why has the percentage changed? 🤔

**Review by jaahon — COMMENTED** [review]
Apart from that test thing I have no comments. Looks good!

File: service/src/integrationTest/kotlin/fi/espoo/evaka/occupancy/OccupancyControllerIntegrationTest.kt (hunk 1 above)
Comment (Gekkio) [reply service/src/integrationTest/kotlin/fi/espoo/evaka/occupancy/OccupancyControllerIntegrationTest.kt:242]: The 3-month period now includes only pure preschool (vs preschool+daycare previously), but the amount of caretakers is the same.

-> lower sum (this child goes 1.0 -> 0.5), lower percentage

Comment (jaahon) [reply service/src/integrationTest/kotlin/fi/espoo/evaka/occupancy/OccupancyControllerIntegrationTest.kt:242]: I stared at this for minutes and still didn't notice that `2.75` was changed to `2.75`. Nevermind 😂

**Review by jaahon — APPROVED** [review]

**Pull request merged**